### ISSUE TITLE
Removing routes on runtime 

### ISSUE THREAD

**Issue by MiroslawSondej** [OrchardCMS/OrchardCore#4586]
I'm maintaining an old application made using OrchardCMS. It's using some very old beta version of Orchard, so I decided to make an upgrade to the RC1 version. I had a problem with this part of Startup.cs code:

```
public class Startup : StartupBase
{
    public override void Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider)
    {
        builder.UseMiddleware<Middleware.ContentsAdminRedirectMiddleware>();

        var oldContentsRoutes = routes.Routes.OfType<Route>().Where(r =>
            r.Defaults["area"] as string == "OrchardCore.Contents"
            && r.Defaults["controller"] as string == "Admin"
        );

        if (oldContentsRoutes.Any())
        {
            foreach (var old in oldContentsRoutes.ToList())
            {
                routes.Routes.Remove(old);

                var action = old.Defaults["action"] as string;
                if(action == "List")
                {
                    action = "CustomList";
                }

                routes.MapAreaRoute(
                    name: old.Name,
                    areaName: "MyCustomEditor.OrchardCore",
                    template: old.RouteTemplate,
                    defaults: new { controller = "ContentsAdmin", action }
                );
            }
        }
    }
}
```

After upgrade, the definition of the "Configure" method is taking "IEndpointRouteBuilder" as a parameter instead of "IRouteBuilder", so I have no way to access and modify existing routes. Is there any way to delete those routes in the "Configure" method on runtime as I did in my code before?

**Comment by hishamco** [comment]
I suggest to refer to the ASP.NET Core docs, coz `IEndpointRouteBuilder` is ASP.NET Core thing, I'm sure you will find a migration guide in the docs

**Comment by sebastienros** [comment]
@jtkech ?

**Comment by Skrypt** [comment]
The router has changed in .NET Core 3.0 I think you will need to use `routes.MapAreaControllerRoute()`

**Comment by jtkech** [comment]
Yes, here not so easy, i could retrieve the related endpoints but was not able to remove them from a read only collection. So here better to use a `DynamicRouteValueTransformer` and maybe, if you also want to generate the right urls in the views, an `IShellRouteValuesAddressScheme`.

See in our `OC.Autoroute` module some concrete implementations and how we register them. See in `AutoRouteTransformer.cs` how we check the current route values and how we return or not a new `RouteValueDictionary` with the updated values.

**Comment by jtkech** [comment]
@Skrypt you're right we now need to use `MapAreaControllerRoute`.

But @MiroslawSondej want to override some routes.

**Comment by jtkech** [comment]
Not tried but another solution would be that in your module startup you do some `MapAreaControllerRoute` related to your custom routes, and that your module registers these route endpoints before the `OC.Contents` module. There is a `ConfigureOrder` property that you can override in a module startup, e.g with a negative value of `-1`.

Need to be tried, maybe it will say that there is a conflict.

**Comment by Skrypt** [comment]
I think it works as long as the route names are not the same.
